improver_tests/set_up_test_cubes.py

```diff
@@ -60,48 +62,131 @@ def construct_xy_coords(ypoints, xpoints, spatial_grid):
             Number of grid points required along the x-axis
         spatial_grid (str):
             Specifier to produce either a "latlon" or "equalarea" grid
+        grid_spacing (int or float):
+            Grid resolution (degrees or metres).
+        domain_corner (list of int or float):
+            Bottom left corner of grid domain [y,x] (degrees or metres). If not
+            provided, a grid is created centred around (0,0).
 
     Returns:
         y_coord, x_coord (tuple):
             Tuple of iris.coords.DimCoord instances
     """
+    if spatial_grid not in GRID_COORD_ATTRIBUTES.keys():
+        raise ValueError("Grid type {} not recognised".format(spatial_grid))
+
+    if grid_spacing is None:
+        if domain_corner is None:
+            y_array, x_array = _default_grid(ypoints, xpoints, spatial_grid)
+        else:
+            raise ValueError("Grid spacing required to setup grid from domain corner.")
+    else:
+        if domain_corner is None:
+            domain_corner = _set_domain_corner(ypoints, xpoints, grid_spacing)
+        y_array, x_array = _create_y_x_arrays(
+            ypoints, xpoints, domain_corner, grid_spacing
+        )
+
+    y_coord = DimCoord(
+        y_array,
+        GRID_COORD_ATTRIBUTES[spatial_grid]["yname"],
+        units=GRID_COORD_ATTRIBUTES[spatial_grid]["units"],
+        coord_system=GRID_COORD_ATTRIBUTES[spatial_grid]["coord_system"],
+    )
+    x_coord = DimCoord(
+        x_array,
+        GRID_COORD_ATTRIBUTES[spatial_grid]["xname"],
+        units=GRID_COORD_ATTRIBUTES[spatial_grid]["units"],
+        coord_system=GRID_COORD_ATTRIBUTES[spatial_grid]["coord_system"],
+    )
+
+    return y_coord, x_coord
+
+
+def _create_y_x_arrays(ypoints, xpoints, domain_corner, grid_spacing):
+    """
+    Creates arrays for constructing y and x DimCoords.
+
+    Args:
+        ypoints (int):
+            Number of grid points required along the y-axis
+        xpoints (int):
+            Number of grid points required along the x-axis
+        domain_corner (list of int or float):
+            Bottom left corner of grid domain [y,x] (degrees or metres)
+        grid_spacing (int or float):
+            Grid resolution (degrees or metres)
+
+    Returns:
+        y_array, x_array (tuple):
+            Tuple of numpy.ndarrays
+    """
+    y_points_array = [domain_corner[0] + i * grid_spacing for i in range(ypoints)]
```

Comment:
This is similar to numpy linspace and/or arange functions. Suggest looking at library functions to check if library functionality is suitable for this case.

Reply:
Couldn't find anything that uses start, step and npoints. 

Would there be any benefit here to doing something like using numpy arange with stop calculated from domain corner, grid spacing and npoints? Possibly a bit more readable...

```py
y_stop = domain_corner[0] + (grid_spacing * ypoints)
x_stop = domain_corner[1] + (grid_spacing * xpoints)

y_array = np.arange(domain_corner[0], y_stop, grid_spacing, dtype=np.float32)
x_array = np.arange(domain_corner[1], x_stop, grid_spacing, dtype=np.float32)
```

Reply:
Yes, calculating the end value and using arange would be better than the current list comprehension. It's a bit more readable and avoids the values being initially generated as a python list of floats then converted to a numpy float32 array.

Reply:
Ok, thank you.

Modified in commit 5986b58f5e860a101919bed79d72aa421f777efa.